### ISSUE TITLE
Propagate settings field error status through tree

### ISSUE THREAD

**Issue by rgov** [foxglove/studio#3555]
Foxglove Studio 1.13.2

If one of my settings fields has an error, but the tree is collapsed, I won't know about it.

It would be nice if the parents showed the error icon and clicking on it revealed the erroring field.

**Comment by defunctzombie** [comment]
@foxymiles @2metres sending your way for UX ideas and implementation.

**Comment by jtbandes** [comment]
I believe our latest thinking is that this would pose a performance problem. Is that right @foxymiles?

**Comment by foxymiles** [comment]
Right. I can't think of a general solution to this problem that doesn't require recursing through all descendants on each render. There might be a pure CSS solution using `has()`.

**Comment by jtbandes** [comment]
Did we have other ideas for how to address the UX issue of not being able to find errors?

**Comment by foxymiles** [comment]
At the moment there's nothing stopping panel authors from handling this themselves explicitly. If we want to solve it in a general way I'd be tempted to try some CSS based `has()` solutions.

**Comment by defunctzombie** [comment]
> performance killing tree traversal on every render on the settings UI side

Only on every update to the settings state not every render?

**Comment by foxymiles** [comment]
> Only on every update to the settings state not every render?

You mean every render of the parent panel? There are many cases where settings gets updated very rapidly like panning the camera in 3d panel or scrubbing a number value or the color picker or even typing quickly in a settings field.

I still think trying something with the `has:` selector is worthwhile.
